KAS-2356 (VAL-254) agenda data loading

MikiDi opened the pull request:
Refactors voor het verbeteren en verduidelijken van de data-loading van agenda-overzicht alsook de scrolbare lijst van punten in het een agenda-detail.
Zie [KAS-2356](https://kanselarij.atlassian.net/browse/KAS-2356)

Needs https://github.com/kanselarij-vlaanderen/kaleidos-project/pull/174 in the backend.

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
refactor: remove unused model property

MikiDi committed:
refactor: rename agendaItemsto the more clear & specific "notas"

MikiDi committed:
refactor: make meeting & agendaitem tracked

MikiDi committed:
refactor: explicit page size inst of relying on disabled backend pagination

MikiDi committed:
refactor: remove unneeded sorts on priority

MikiDi committed:
refactor: remove unused empty-check

MikiDi committed:
refactor: concurrency decorator syntax

MikiDi committed:
refactor: rename agendaitems to notas for clarity

MikiDi committed:
refactor: agenda-overview to glimmer component

MikiDi committed:
feat: component for agenda item group header

MikiDi committed:
refactor: agenda overview grouping

MikiDi committed:
refactor: only modify items prios that need modifying

MikiDi committed:
refactor: make agendaitem reordering DDAU

MikiDi committed:
style: remove spacing

MikiDi committed:
feat: remove in-viewport hooks for details loading

MikiDi committed:
refactor: remove dead code

MikiDi committed:
refactor: remove unused argument

MikiDi committed:
refactor: disregard isEditingOverview for now

MikiDi committed:
refactor: make formally ok status hiding DDAU

MikiDi committed:
refactor: remove unused alias

MikiDi committed:
refactor: remove non-existant css property

MikiDi committed:
refactor: remove duplicate getter with unused props

MikiDi committed:
refactor: load subcase through task

MikiDi committed:
refactor: load newsletter visibility in overview

MikiDi committed:
fix: automatically create agenda-item-treatment for approval item

MikiDi committed:
fix: add fallback for when no treatment exists

MikiDi committed:
refactor: remove unused selectedAgendaitem prop

MikiDi committed:
refactor: remove unused aboutToDelete prop

MikiDi committed:
refactor: initialize props in controller

MikiDi committed:
refactor: remove unused if-clause

MikiDi committed:
refactor: clean up if conditions

MikiDi committed:
refactor: remove unused if clause

MikiDi committed:
refactor: restructure if clause + rename action for clarity

MikiDi committed:
refactor: simplify properties related to formally ok

MikiDi committed:
refactor: add this (glimmer component)

MikiDi committed:
refactor: remove useless isViewer condition

MikiDi committed:
refactor: always show separator for now. Granular loading states later

MikiDi committed:
refactor: angle-bracket notation

MikiDi committed:
refactor: simplify document visibility logic

MikiDi committed:
refactor: simplify document visibility logic

MikiDi committed:
refactor: make loading state specific to start-pub button

MikiDi committed:
refactor: css highlighting logic

MikiDi committed:
refactor: implement document list rendering with new components

MikiDi committed:
Merge branch 'feature/VAL-254-agenda-data-loading' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/VAL-254-agenda-data-loading

erikap committed:
Add missing action to toggle more documents on agenda overview

erikap committed:
Correct view on agenda in read-only mode

- Add missing group headers
- Add approval notas on top

erikap committed:
Upgrade ember-sortable to v2.x

erikap committed:
Merge sortable and non-sortable agenda by disabling drag-and-drop

erikap committed:
Add required include for first render in agendaitems route

A limited set of fields for mandatees, pieces and document-containers
is required for an initial render. We only include those fields in the
model hook.

erikap committed:
Fix linting errors

erikap committed:
Refactor SortableGroup to modifier syntax on agenda overview

erikap committed:
Replace deprecated usage of agendaitem number with priority

erikap committed:
Bump ember-in-viewport addon

erikap committed:
Lazy load secondary data only on viewport enter in agenda overview

erikap committed:
Update regular notas on search in agenda in agenda overview

erikap committed:
Fix initial filter in agenda overview if filter param is set

erikap committed:
Add toggle to only view modified agendaitems in agenda overview

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
linting

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
refactor: avoid in-place sorting of mandatees + pre-calc one-time ops

MikiDi committed:
refactor: document-badge component

MikiDi committed:
feat: preview method for loading new (highlighted) documents more granulary

MikiDi committed:
refactor: implement refactored document-badge

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
Merge branch 'development' into feature/VAL-254-agenda-data-loading

MikiDi committed:
style: concurrency decorators + indenting on template for search bar

MikiDi committed:
fix: revert 8407be526e1d0e4187fc8c8b0c7934fd58f9757f . @value != value

MikiDi committed:
refactor: use non-deprecated version of contains helper

MikiDi committed:
feat: start using new doc compare endpoint

MikiDi committed:
fix: use singularized model name for store access

MikiDi committed:
fix: push the ember-data object to array instead of the pojo

MikiDi committed:
fix: modifiers don't seem to work on components

MikiDi committed:
more stuff for start using new compare endpoint

MikiDi committed:
refactor: do away with unused properties and aliases

MikiDi committed:
feat: formally ok selector to glimmer

MikiDi committed:
refactor: integrate new formally ok status selector

MikiDi committed:
feat: reinstate toggling logic for overview editing

MikiDi committed:
refactor: overview mode switching: make both links button-links

MikiDi committed:
fix: forgot to mark property as tracked

MikiDi committed:
fix: forgot to integrate new formally ok selector for all isntances

MikiDi committed:
refactor: specify this

MikiDi committed:
fix: don't drag between different item categories

MikiDi committed:
refactor: don't keep a separate group for approval items

MikiDi committed:
style: linting

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
fix: add & refine loading states for agenda-overview loading

MikiDi committed:
chore: installed ember-concurrency and updagraded ember-concurrency-decorators

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
chore: sync package-lock

MikiDi committed:
feat: only one viewport handler per component improves scroll perf

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
Merge branch 'feature/VAL-254-agenda-data-loading' into feature/VAl-254-agenda-data-loading--one-viewport-handler-per-component

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
feat: add loading state for newsletter visibility

MikiDi committed:
chore: sync package-lock

MikiDi committed:
Merge branch 'feature/VAl-254-agenda-data-loading--one-viewport-handler-per-component' into feature/VAL-254-agenda-data-loading

MikiDi committed:
feat: tweak lazy loading timeout

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
Merge branch 'feature/VAL-225-submission-activity' into feature/VAL-254-agenda-data-loading

MikiDi committed:
style: consisten route & controller class naming

MikiDi committed:
refactor: remove unnecessary set of model + unnecessary setter use

MikiDi committed:
fix: document-card-list after document-card refactor merge

MikiDi committed:
refactor: add index route for agenda-item-overview

MikiDi committed:
refactor: replace a-tag by button-link

MikiDi committed:
refactor: remove unused spacerClass

MikiDi committed:
refactor: code reuse for agendaitem-overview vs detail pane

MikiDi committed:
refactor: remove unused class

MikiDi committed:
refactor: remove unused properties + cleaner css class composition

MikiDi committed:
refactor: alias isRetracted

MikiDi committed:
refactor: make "isActive" DDAU

MikiDi committed:
refactor: remove unused select-agenda-item action-handler

MikiDi committed:
refactor: remove unused definitions

MikiDi committed:
doc: clean up component argument doc

MikiDi committed:
refactor: implement inheritance for sidebar/overview-item lazy loading

MikiDi committed:
feat: loading state for sidebar while grouping items

MikiDi committed:
fix: degraded performance issue due to tracked query parameters
https://github.com/emberjs/ember.js/issues/18715#issue-557737948

MikiDi committed:
feat: DDAU toggle "new items only" for sidebar

MikiDi committed:
doc: document component argument

MikiDi committed:
refactor: move "isDesignAgenda" calc to parent class

MikiDi committed:
refactor: make show-formally-ok-status DDAU

MikiDi committed:
feat: hide status icon when not yet loaded

MikiDi committed:
fix: always show border-bottom for annoucement title

MikiDi committed:
refactor: remove unused service injections

MikiDi committed:
refactor: remove unused aliases and service injections

MikiDi committed:
Merge branch 'development' into feature/VAL-254-agenda-data-loading

MikiDi committed:
chore: move around files after moving away from pods components

MikiDi committed:
Merge branch 'development' into feature/VAL-254-agenda-data-loading

MikiDi committed:
chore: component file-move related changes

MikiDi committed:
doc: update document-list component doc after making it renderless

ValenberghsSven committed:
Merge branch 'development' of https://github.com/kanselarij-vlaanderen/kaleidos-frontend into feature/VAL-254-agenda-data-loading

ValenberghsSven reviewed (commented):
Voorlopig gevonden issues:

- Gele balk "geen toekenning" mist, al zeker tussen het verslag v/d vorige vergadering en het eerst volgende agendapunt zonder ministers
- Wijzigen van ministers triggert geen change of refresh in de sidebar
- Meerdere ministers laden slecht, er staan lege rijen (1e minister zichtbaar, 2e lege rij), zodra je edit om beleidsvelden te wijzigen verschijnt de informatie (loading issue ?)
- Verslepen van agendapunten is terug het hele object ipv die icoontjes. Die icoontjes gebruiken om te verslepen is er gekomen omdat het te vaak per ongeluk versleept werd. Waarom zijn die icoontjes eruit ?
- Toon van aangepaste punten voor nieuwe agendapunten werkt pas na refresh
- Toon van aangepaste punten werkt niet meer voor nieuwe documenten op bestaande agendapunten
- Toevoegen van agendapunten op overzicht triggert geen refresh, manueel refresh is nodig om data te hebben

ValenberghsSven committed:
VAL-254 cypress fix changed XHR calls

MikiDi committed:
Merge branch 'development' into feature/VAL-254-agenda-data-loading

MikiDi committed:
fix: included mandatees for reload on detail-route
Ember-data thinks to have mandatees loaded, but we only load necessary
fields in the overview for efficfiency.

MikiDi committed:
fix: @ instead of this for currentAgenda

MikiDi committed:
refactor: expose showing of the drag handle DDAU

MikiDi commented:
> * Gele balk "geen toekenning" mist, al zeker tussen het verslag v/d vorige vergadering en het eerst volgende agendapunt zonder ministers

:grey_question: Check. Punten bovenaan de agenda zonder toegewezen ministers (meestal enkel punt goedkeuring notulen) tonen geen lege gele balk. Da's iets generieker dan vroeger. Het werkt een stuk vlotter als we het goedkeuringspunt op dezelfde manier als elk ander punt kunnen behandelen. Best nog te verifiëren inderdaad. Groepen verder in de agenda zonder ministers krijgen wel steeds de "geen toekenning"-balk

> * Wijzigen van ministers triggert geen change of refresh in de sidebar

:red_circle: Check. TODO. De minister-component moet DDAU gerefactored worden om de agendapunt-groepering te kunnen herberekenen wanneer een minister toegevoegd wordt.

> * Meerdere ministers laden slecht, er staan lege rijen (1e minister zichtbaar, 2e lege rij), zodra je edit om beleidsvelden te wijzigen verschijnt de informatie (loading issue ?)

:green_circle: Check. Fixed. Zie https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/778/commits/5b677245b12609bc7115a8af4a2c100e4c1105dd voor meer details.

> * Verslepen van agendapunten is terug het hele object ipv die icoontjes. Die icoontjes gebruiken om te verslepen is er gekomen omdat het te vaak per ongeluk versleept werd. Waarom zijn die icoontjes eruit ?

:green_circle: Check. Fixed. Zie 2 laatste commits voor meer detail.

> * Toon van aangepaste punten voor nieuwe agendapunten werkt pas na refresh

:grey_question: Wat bedoel je juist met "Toon van aangepaste punten"?

> * Toon van aangepaste punten werkt niet meer voor nieuwe documenten op bestaande agendapunten

:grey_question: Idem als hierboven.

> * Toevoegen van agendapunten op overzicht triggert geen refresh, manueel refresh is nodig om data te hebben

:red_circle: Check. TODO. Idem als 2 punten hierboven. Agenda-acties DDAU. Potentieel grote impact.

ValenberghsSven commented:
@MikiDi 
Toon aangepaste punten, de functionaliteit waarmee je kan zien welke agendapunten nieuw zijn, of welke agendapunten nieuwe documenten / versies hebben sinds vorige goedgekeurde agenda
Rechtsboven de agendapunten in overzicht, naast wijzigen knop van formeel ok's
boven de sidebar in detail
![image](https://user-images.githubusercontent.com/22245223/110930618-a03a3880-8329-11eb-958b-ef1c508193ce.png)
![image](https://user-images.githubusercontent.com/22245223/110930752-c9f35f80-8329-11eb-9f71-59e327c00008.png)

MikiDi committed:
refactor: session-agenda-nav to glimmer component

ValenberghsSven commented:
Correctie daarop, documenten toevoegen werkt wel om changes te zien, maar pas na een refresh
Gisteren leek het dat nieuwe docs niet werkten, ook niet na refresh

MikiDi committed:
refactor: use LinkTo instead of action for navigation

MikiDi committed:
refactor: get rid of selectedAgendaItem global state

MikiDi committed:
refactor: move agenda-item-case component to agenda-item detail template

MikiDi committed:
refactor: move minister-modal to more sensible location

MikiDi committed:
refactor: rename mandatee domain selector for clarity

MikiDi committed:
refactor: build new DDAU mandatees-domains panel

MikiDi committed:
refactor: mandatees-domains-selector to octane

MikiDi committed:
feat: separate component for domain/field-select checkboxes

MikiDi committed:
refactor: implement domains-fields-selector in domains-fields-modal

MikiDi committed:
refactor: move editing-state down in edit sub-component

MikiDi committed:
refactor: integrate domains-selector-modal in edit-panel

MikiDi committed:
feat: renderless component for filtering fields related to mandatee

MikiDi committed:
fix: add cancel action

MikiDi committed:
feat: integrate renderless fields filter for edit panel

MikiDi committed:
fix: remove action handler

MikiDi committed:
chore: remove no longer valid comment & log

MikiDi committed:
fix: fallback for empty arrays

MikiDi committed:
refactor: add wiring for changed fields

MikiDi committed:
fix: call task with perform

MikiDi committed:
fix: remove unused constructor task

MikiDi committed:
chore: add ember-composable-helpers for array include helper

MikiDi committed:
chore: re-order package.json

MikiDi committed:
refactor: rename & move ise-codes-panel to something more sensible

MikiDi committed:
refactor: more logical interface for ise-code panel

MikiDi committed:
refactor: apply condition only on dependant portion

MikiDi committed:
refactor: move contents of subcase-case component to route

MikiDi committed:
fix: read from buffer, not initialization args

MikiDi committed:
fix: only one collumn provided for domains & fields

MikiDi committed:
refactor: replace subcase-mandatees component by the generic one

MikiDi committed:
fix: load fields in agenda-item index route

MikiDi committed:
refactor: add handler for mandatee edits to subcase controller

MikiDi committed:
fix: test on the right array if a minister is the submitter

MikiDi committed:
fix: pass down array as supposed to

MikiDi committed:
fix: await query

MikiDi committed:
fix: mark fields as tracked

MikiDi committed:
fix: test the correct array to determine if fields have been removed

MikiDi committed:
fix: adapt controller state without reload

MikiDi committed:
fix: correctly load side-data before the template renders

MikiDi committed:
fix: handle mandatee changes correctly on both agenda-item/subcase-sides

MikiDi committed:
fix: fetch and interface-update government-fields after modification

MikiDi committed:
fix: fetch and interface-update government-fields after modification
On the subcase-side.

MikiDi committed:
fix: a better try at displaying a list of fields

MikiDi committed:
refactor: remove no-longer-used minister utils

MikiDi committed:
chore: remove no longer used components

MikiDi committed:
Merge branch 'development' into feature/KAS-2398-minister-DDAU

MikiDi committed:
Merge branch 'development' into feature/VAL-254-agenda-data-loading

MikiDi committed:
style: fix linting error

MikiDi committed:
fix: removing unused reloadModel, allows bubbling
This fixes an issue where adding a new agenda-item would not trigger a
re-load of the agenda-items-route (the one above the index). The fact
that this fixes the aforementioned issue, is more a side-effect than
anything else. We should figure out a more reliable way to reload the
model of a given route while awaiting implementation of
https://github.com/emberjs/rfcs/blob/master/text/0631-refresh-method-for-router-service.md

MikiDi committed:
feat: pass the newly created agenda-items in the action up

MikiDi committed:
Merge branch 'development' into feature/VAL-254-agenda-data-loading

MikiDi commented:
@ValenberghsSven Klopt volgende stelling?

> Ivm klikken op "Toon aangepaste punten". Punten worden als "aangepast" beschouwd als ze volledig nieuw zijn, óf nieuwe (versies van) documenten hebben. Het "bord met plusje"-icoontje heeft een tooltip "nieuw agendapunt" en wordt énkel getoond als een punt vollédig nieuw is op de agenda en niét wanneer er enkel nieuwe documenten zijn. Wanneer "Toon aangepaste punten" actief is, kunnen dus zowel kaartjes met als zonder "bord met plusje"-icoontje getoond worden. Wanneer kaartjes geen "bord met plusje"-icoontje hebben, zullen ze wel nieuwe (groene) documenten tonen.

ValenberghsSven committed:
VAL-254 fix some tests, link to detail fixed

ValenberghsSven committed:
Merge branch 'feature/VAL-254-agenda-data-loading' of https://github.com/kanselarij-vlaanderen/kaleidos-frontend into feature/VAL-254-agenda-data-loading

ValenberghsSven commented:
> @ValenberghsSven Klopt volgende stelling?
> 
> > Ivm klikken op "Toon aangepaste punten". Punten worden als "aangepast" beschouwd als ze volledig nieuw zijn, óf nieuwe (versies van) documenten hebben. Het "bord met plusje"-icoontje heeft een tooltip "nieuw agendapunt" en wordt énkel getoond als een punt vollédig nieuw is op de agenda en niét wanneer er enkel nieuwe documenten zijn. Wanneer "Toon aangepaste punten" actief is, kunnen dus zowel kaartjes met als zonder "bord met plusje"-icoontje getoond worden. Wanneer kaartjes geen "bord met plusje"-icoontje hebben, zullen ze wel nieuwe (groene) documenten tonen.

@MikiDi 
Klopt.
Het is ook zo dat die aangepaste punten moeten updaten zonder een volledige page refresh.
Dus we moeten bij uitvoeren van bepaalde acties (toevoegen nieuw agendapunt, toevoegen docs) die api call naar de agenda-sort-service doen.
Met de nieuwe data loading manier moeten we dan vermoedelijk het model refreshen ?

ValenberghsSven commented:
Extra issue dat ik heb opgemerkt, als je een agendapunt verwijdert 
![image](https://user-images.githubusercontent.com/22245223/112150649-9f828b80-8be0-11eb-870a-a8a1c8875cb9.png)

Blijft het agendapunt staan in de sidebar, de sidebar reload niet.
Als je dan klikt op dat agendapunt:
![image](https://user-images.githubusercontent.com/22245223/112150724-b6c17900-8be0-11eb-8de6-62df86909112.png)

ValenberghsSven commented on app/pods/agenda/agendaitems/route.js:
```diff
@@ -1,57 +1,80 @@
     });
 
-    const announcements = agendaitems.filter((agendaitem) => agendaitem.showAsRemark);
+    if (params.showModifiedOnly) {
+      // agendaService.agendaWithChanges is called by parent route
+      const matchingIds = this.agendaService.addedAgendaitems;
+      agendaitems = agendaitems.filter((ai) => matchingIds.includes(ai.id));
+    }
```
@MikiDi Hier zit het probleem dat documenten niet aanzien worden al changes
hier wordt enkel gecheckt of het agendapunt nieuw is.
In het agendaitem model is er een property 'hasChanges'. Kunnen we die hier gebruiken ?

MikiDi replied:
Dank voor je analyse. Ik moet met de informatie uit https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/778#issuecomment-802915649 nog éen en ander in praktijk zetten. Zeker voor de toegevoegde/verwijderde punten zal een model refresh idd aan de orde zijn. `hasChanges` toevoegen op agendapunt lijkt me niet waar we naartoe willen. `hasChanges` is immers niet echt een eigenschap van het punt. Het hangt er van af in welke context (op welke agenda) je het punt bekijkt of het nu al dan niet `changed` is. Ik pas het aan.

ValenberghsSven commented:
De indieningsactiviteit aanmaken geeft een popup 'translation for "submission-activitie" not found'
Dit komt uit de application.js adapater, daar gaan we type slicen en singular maken, maar dat werkt dus niet voor dit type.
Willen we überhaupt die popup tonen ? dat doen we ook niet voor agendering activiteit en publicatie activiteiten.

De gebruiker heeft volgens mij geen boodschap aan een pop-up "Er is een indiendingsactiviteit aangemaakt"

MikiDi commented:
> De indieningsactiviteit aanmaken geeft een popup 'translation for "submission-activitie" not found'
> Dit komt uit de application.js adapater, daar gaan we type slicen en singular maken, maar dat werkt dus niet voor dit type.
> Willen we überhaupt die popup tonen ? dat doen we ook niet voor agendering activiteit en publicatie activiteiten.
> 
> De gebruiker heeft volgens mij geen boodschap aan een pop-up "Er is een indiendingsactiviteit aangemaakt"

Klopt. Dat willen we in de gehele applicatie herwerken (ik wist niet dat er al enige configuratie in zat die maakt dat de popup voor sommige resources niet getoond wordt). We willlen bevestiging op basis van een gebruikersinteractie, niet op basis van het opslaan van (soms technisch geïnspireerde) entitetiten waar de gebruiker geen boodschap aan heeft ... [KAS-1412](https://kanselarij.atlassian.net/browse/KAS-1412): TODO. Maar dat is hier niet in scope, toch (ook al aanwezig op dev?)?

MikiDi committed:
Merge branch 'development' into feature/VAL-254-agenda-data-loading

MikiDi committed:
fix: refresh parent model when removing an agenda-item

ValenberghsSven commented:
> > De indieningsactiviteit aanmaken geeft een popup 'translation for "submission-activitie" not found'
> > Dit komt uit de application.js adapater, daar gaan we type slicen en singular maken, maar dat werkt dus niet voor dit type.
> > Willen we überhaupt die popup tonen ? dat doen we ook niet voor agendering activiteit en publicatie activiteiten.
> > De gebruiker heeft volgens mij geen boodschap aan een pop-up "Er is een indiendingsactiviteit aangemaakt"
> 
> Klopt. Dat willen we in de gehele applicatie herwerken (ik wist niet dat er al enige configuratie in zat die maakt dat de popup voor sommige resources niet getoond wordt). We willlen bevestiging op basis van een gebruikersinteractie, niet op basis van het opslaan van (soms technisch geïnspireerde) entitetiten waar de gebruiker geen boodschap aan heeft ... [KAS-1412](https://kanselarij.atlassian.net/browse/KAS-1412): TODO. Maar dat is hier niet in scope, toch (ook al aanwezig op dev?)?

Dan kunnen we nu gewoon die 'submission-activities' toevoegen als genegeerd type, en is het probleem opgelost voor dit ticket

MikiDi committed:
feat: make filtering on "modified" items work like before

MikiDi committed:
feat: redirect to one of visible items when item dissapears after filtering

MikiDi committed:
refactor: implement fetching new agenda-items + on model at route level

MikiDi committed:
fix: handle case where no fields are passed to domains-fields-for-mandatee

MikiDi committed:
feat: default submitter

MikiDi committed:
fix: correctly load submitter + interface update after change

MikiDi committed:
fix: update ISE-codes in interface after changing mandatees

MikiDi committed:
fix: load mandatees in ordered fashion

MikiDi committed:
feat: include submitter in request for agenda-item

MikiDi committed:
fix: mandatees, submitter & ise-code interface-state on subcase

MikiDi committed:
doc: remove stray console.log and add future work comment

ValenberghsSven committed:
KAS-2398 ensure controllers have correct data, cleanup & unify selectors, fix tests

ValenberghsSven committed:
KAS-2398 no it.only

ValenberghsSven committed:
cypress test selectors with angle brackets need a value to exist after prod build

ValenberghsSven committed:
KAS-2398 fix final tests, remove dead code for setting confidentiality

ValenberghsSven committed:
VAL-254 revive selector, change data loading after delete, wait for data loading in some cases

MikiDi committed:
fix: adapt item-related fields when a mandatee gets removed

MikiDi committed:
Merge branch 'feature/KAS-2398-minister-DDAU' into feature/VAL-254-agenda-data-loading

ValenberghsSven committed:
VAL-254 fix last test, wait for data loading

ValenberghsSven committed:
VAL-254 cypress: checking loading modal for data loading issues

MikiDi committed:
refactor: move AgendaitemGroup to util

MikiDi committed:
doc: document AgendaitemGroup class

MikiDi committed:
refactor: prepare sidebar/overview-components for taking nota-groups

MikiDi committed:
refactor: feed nota-groups into sidebar/overview-component from ctrllr

MikiDi committed:
refactor: re-calculate item-grouping when modifying mandatees

MikiDi committed:
Merge branch 'feature/VAL-254-agenda-data-loading' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/VAL-254-agenda-data-loading
